src-docs/src/components/guide_page/guide_page_header.tsx

```diff
@@ -32,29 +35,72 @@ export const GuidePageHeader: React.FunctionComponent<GuidePageHeaderProps> = ({
 }) => {
   const isMobileSize = useIsWithinBreakpoints(['xs', 's']);
 
-  function renderLogo() {
+  const logo = useMemo(() => {
     return (
       <EuiHeaderLogo iconType={logoEUI} href="#/" aria-label="EUI home">
         Elastic UI
       </EuiHeaderLogo>
     );
-  }
+  }, []);
 
-  function renderVersion() {
+  const [isVersionPopoverOpen, setIsVersionPopoverOpen] = useState(false);
+  const versionBadge = useMemo(() => {
     const isLocalDev = window.location.host.includes('803');
-
     return (
       <EuiBadge
-        href="#/package/changelog"
-        aria-label={`Version ${pkg.version}, View changelog`}
+        onClick={() => setIsVersionPopoverOpen((isOpen) => !isOpen)}
+        onClickAriaLabel={`Version ${pkg.version}. Click to switch versions`}
         color={isLocalDev ? 'accent' : 'default'}
       >
         {isLocalDev ? 'Local' : `v${pkg.version}`}
       </EuiBadge>
     );
-  }
+  }, []);
+  const versionSwitcher = useMemo(() => {
+    return (
+      <EuiPopover
+        isOpen={isVersionPopoverOpen}
+        closePopover={() => setIsVersionPopoverOpen(false)}
+        button={versionBadge}
+        repositionOnScroll
+        panelPaddingSize="xs"
+      >
+        <FixedSizeList
+          className="eui-yScroll"
+          itemCount={euiVersions.length}
+          itemSize={24}
+          height={200}
+          width={120}
+          innerElementType="ul"
+        >
+          {({ index, style }) => {
+            const version = euiVersions[index];
+            return (
+              <EuiListGroupItem
+                style={style}
+                size="xs"
+                label={`v${version}`}
+                href={`https://eui.elastic.co/v${version}/`}
+                extraAction={{
+                  'aria-label': 'View release',
```

Comment:
Let's add the release number to these `aria-labels`. It'll improve the experience for users who open their screen reader's Links menu to hear "View release eighty-nine point one point zero" instead of just "View release".

```suggestion
                  'aria-label': `View release ${version}`,
```